### ISSUE TITLE
KubePodCrashLooping timewindow too big for exponential kubelet restart behavior

### ISSUE THREAD

**Issue by gertjangaillet** [kubernetes-monitoring/kubernetes-mixin#526]
The current time window used in the expression for KubePodCrashLooping is 5 minutes. It seems it was reduced from 15 minutes to 5 minutes on purpose with [this commit](https://github.com/kubernetes-monitoring/kubernetes-mixin/commit/e962182524fdb97a1ce27755ec7af9ad918cf652#diff-c511c672e0359c4c2fbe255e8cc894caf1c05b8f3a0db74edc1c38e797e23afc), of which I can't find the context or reason.

However, as far as I know the kubelet restarts pods with an exponential delay going up to 5 minutes, and this limit of 5 minutes is reached already within 10 minutes in our experience. So by the time the alert would reach the Firing status, the delay of 5 minutes is already reached. Typically containers can take anywhere from a few seconds to a few minutes to start, adding to the 5 minutes restart delay of the kubelet. That means that within 10 minutes from the first restart, we already reach a situation where the pod does restart but more than 5 minutes after the previous restart happens.

As a result, at some points in time no restart happened the past 5 minutes, because the previous restart happened for instance 6 minutes ago, and the alert is cleared. A few moments later, a restart happens and the alert is back in Pending status, however a few minutes later it is cleared again. 

To summarize, in most situations, this alert simply never fires.

Can we therefore increase the time window of this check again, from 5 minutes to for instance 10 minutes?

**Comment by brancz** [comment]
I think trying 10min sounds good to me.

**Comment by gertjangaillet** [comment]
Ok @brancz - I have created a PR - can you have a look please?

**Comment by povilasv** [comment]
PR merged
